rust/src/quic/crypto.rs

```diff
@@ -113,9 +111,8 @@ impl PacketKey {
         }
         let tag_pos = payload.len() - AES128_TAG_LEN;
         let (buffer, tag) = payload.split_at_mut(tag_pos);
-        let taga = GenericArray::from_slice(tag);
         self.key
-            .decrypt_in_place_detached(GenericArray::from_slice(&nonce), header, buffer, taga)
+            .decrypt_in_place_detached((&nonce).into(), header, buffer, (&*tag).into())
```

Comment:
Confused by `&*` notation in rust. What does it mean ?

Reply:
Type coercion. `tag` as is an `&mut [u8]`, `decrypt_in_place_detached` requires a `Tag`, but the into is not implemented for `&mut [u8]`, but is for `&[u8]`. So essentially casting it to something there is from/into trait to tag for.